### ISSUE TITLE
settings.json schema flags color names as invalid

### ISSUE THREAD

**Issue by TonyGravagno** [Gruntfuggly/todo-tree#606]
From ReadMe:

> Note: Foreground and background colours can be specified using [HTML/CSS colour names](https://en.wikipedia.org/wiki/Web_colors) (e.g. "Salmon"), RGB hex values (e.g. "#80FF00"), RGB CSS style values (e.g. "rgb(255,128,0)" or colours from the current theme, e.g.

Error from current VSCode: 

> Invalid color format. Use #RGB, #RGBA, #RRGGBB or #RRGGBBAA.

settings.json:
```json
"QUESTION": {
            "foreground": "blue",
            "background": "goldenrod",
            "icon": "question",
            "iconColour": "goldenrod",
            "gutterIcon": true
        },
```

Screenshot:
![image](https://user-images.githubusercontent.com/499473/153968503-98c47f16-6804-403c-8547-7bf0fa742aef.png)

Thanks.

**Comment by coppinri** [comment]
I've got the same query/issue.

I think this is more of an issue with some changes that have come down from VSCode than for Todo-Tree. Todo-Tree seems to work find dispite the warning.

If anyone knows how to disable this particular warning, that would be great.

I suppose there's always the option to acquiesce and just use the #RRGGBB values.

**Comment by gerardlt** [comment]
I've just come across the same thing.

Seeing the issue was only raised two days ago and already two more of us have felt it worth commenting on, I'd guess there;s been a recent change in this area.

A quick web search for the warning message got me only two links (surprisingly, given the generic message):

- This comment from 2017: https://github.com/microsoft/vscode/issues/23016#issuecomment-291094997
- And [this code file](https://github.com/microsoft/vscode-json-languageservice/blob/b54c913d69df6fb2bd5419cc8e8f153c6857863c/src/parser/jsonParser.ts). The relevant change [dates from Feb 2020](https://github.com/microsoft/vscode-json-languageservice/commit/2ea5ad3d2ffbbe40dea11cfe764a502becf113ce).

But when I then searched for the message in the vscode repo itself (on github), I additionally found [referencesValidation.ts](https://github.com/microsoft/vscode/blob/ea47790d3982f66a48fb5eed678bee95c29b52d3/src/vs/workbench/services/preferences/common/preferencesValidation.ts) for which the relevant commit was https://github.com/microsoft/vscode/commit/2445428308d74f1eedf73715809dca2d37e21c6d - which also dates from 2020 (November).

So neither of them suggest a recent change. ¯\_(ツ)_/¯

**Comment by TonyGravagno** [comment]
Agreed - the issue is not with this extension. I see the same issue in settings for other extensions. I'm poking around at jsonParser.ts and other vectors as well.

Let's leave this here for discussion and I'll close it if/when we find a culprit we can point at.

BTW, from looking around, this extension has a Ton of util code for validating JSON (colors) and so much else. It's seriously rich FOSS.

**Comment by TonyGravagno** [comment]
All : I think this is our problem and not a bug. I don't see any reference that indicates that a color is accepted in VSCode Settings.

[This comment](https://github.com/microsoft/vscode/issues/34663#issuecomment-331818604) by a VSCode insider @aeschli says the decision was made prior to 2017 to only support RGB.

The [schemas](https://github.com/microsoft/vscode/search?q=color-hex) show all colors are `"format": "color-hex"`.

The documentation for [Color formats](https://code.visualstudio.com/api/references/theme-color#color-formats) says:
> Color values can be defined in the RGB color model with an alpha channel for transparency. As format, the following hexadecimal notations are supported: #RGB, #RGBA, #RRGGBB and #RRGGBBAA. 

The VSCode [JSON language server doc](https://github.com/microsoft/vscode/blob/94c9ea46838a9a619aeafb7e8afd1170c967bb55/extensions/json-language-features/server/README.md) (also written/edited by @aescchli) says:
> All values marked with "format": "color-hex" (VSCode specific, non-standard JSON Schema extension) are considered color values. **The supported color formats are #rgb[a] and #rrggbb[aa].**

Like everyone else here, I've used color names in various places, but I can't say with certainty that I've ever used a color name in VSCode settings.json - and actually had it work. I think I/we are simply wrong to have expected this to work. 😵‍ 

-----

I'm revising this ticket to be a simple error in the [Readme](https://github.com/Gruntfuggly/todo-tree) where it says:
> Note: Foreground and background colours can be specified using [HTML/CSS colour names](https://en.wikipedia.org/wiki/Web_colors) (e.g. "Salmon"), RGB hex values (e.g. "#80FF00"), RGB CSS style values (e.g. "rgb(255,128,0)" or colours from the current theme, e.g. peekViewResult.background. See [Theme Color](https://code.visualstudio.com/api/references/theme-color) for the details. Hex and RGB values can also have an alpha specified, e.g. "#ff800080" or "rgba(255,128,0,0.5)".

Correct text should be:
> Note: Foreground and background colours can be specified using RGB hex values (e.g. "#80FF00"), and RGB with an alpha specified, e.g. "#ff800080". Full format list: #RGB, #RGBA, #RRGGBB or #RRGGBBAA. See [Theme Color](https://code.visualstudio.com/api/references/theme-color) for the details.

Agreed?

**Comment by Gruntfuggly** [comment]
It's sort of my fault - as @TonyGravagno says, the colour settings in the package.json are marked as `"format": "color-hex"` which allows the json settings view to display the colour swatch next to them (and a colour picker when clicked). However, this only works when an RGB.

The extension is flexible in that it allows colour names to be used in place which are supported in the decorations API, but they obviously don't fit the format, so vscode complains.

I could make them `"format": "string"` which will get rid of the warnings, but prevent the colour swatch being displayed.

**Comment by Gruntfuggly** [comment]
I'm also going to add a command to convert named colours to RGB, so users can use it once they're happy with their colour scheme.

**Comment by TonyGravagno** [comment]
A very productive thread. Thank you very much, Nigel!

**Comment by acgc99** [comment]
Still having this issue four years later, any update on this?
```
    "todo-tree.highlights.defaultHighlight": {
        "type": "tag",
        "foreground": "editorInfo.foreground",  # Invalid color format. Use #RGB, #RGBA, #RRGGBB or #RRGGBBAA.
        "background": "#00000000"
    },
```
